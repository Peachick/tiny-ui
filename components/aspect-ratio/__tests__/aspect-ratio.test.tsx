import React from 'react';
import { render } from '@testing-library/react';
import AspectRatio from '../index';

describe('<AspectRatio />', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<AspectRatio />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render children', () => {
    const { container } = render(<AspectRatio>children</AspectRatio>);
    expect(container.firstChild).toHaveStyle({ width: '100%' });
    expect(container).toContainHTML('children');
  });

  it('should render correct ratio container', () => {
    const { container } = render(<AspectRatio ratio={4 / 3} />);
    expect(container.querySelector('.ty-aspect-ratio__padding')).toHaveStyle({ paddingTop: '75%' });
  });

  it('should render correct width', () => {
    const { container } = render(<AspectRatio width={20} />);
    expect(container.firstChild).toHaveStyle({ width: '20px' });
  });
});
