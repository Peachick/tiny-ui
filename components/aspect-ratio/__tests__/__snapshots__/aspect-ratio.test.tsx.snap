// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AspectRatio /> should match the snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ty-aspect-ratio",
      "style": "width:100%",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ty-aspect-ratio__padding",
          "style": "padding-top:100%",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Object {
          "attribs": Object {
            "class": "ty-aspect-ratio__inside",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
      Object {
        "attribs": Object {
          "class": "ty-aspect-ratio__inside",
        },
        "children": Array [],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Object {
          "attribs": Object {
            "class": "ty-aspect-ratio__padding",
            "style": "padding-top:100%",
          },
          "children": Array [],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
